src/components/Feeds/Feeds.jsx

```diff
@@ -20,19 +21,46 @@ const Feeds = () => {
     const { userDetails, isUserDetailLoading, isUserDetailError, userDetailMessage } = useSelector(
         (state) => state.userDetail,
     );
+    const { user } = useSelector((state) => state.auth);
 
     useEffect(() => {
         if (isFeedError) console.log(feedMessage);
         if (isUserDetailError) console.log(userDetailMessage);
 
         dispatch(getAllFeeds());
         dispatch(getAllUserDetails());
-
+        user && dispatch(getFollowData(user._id));
         return () => {
             dispatch(feedReset());
             dispatch(userDetailReset());
+            dispatch(reset());
         };
     }, [dispatch]);
+    const [showOnlyFollowingposts, setShowOnlyFollowingPosts] = useState(false);
+
+    const renderFollowingFilterButtons = () => (
+        <>
+            <FilterButton
+                style={{
+                    background: showOnlyFollowingposts === false ? "#FF6B08" : "",
+                    color: showOnlyFollowingposts === false ? "#0A0A0A" : "",
+                }}
+                onClick={() => setShowOnlyFollowingPosts(false)}
+            >
+                All
+            </FilterButton>
+            <FilterButton
+                style={{
+                    background: showOnlyFollowingposts === true ? "#FF6B08" : "",
```

Comment:
Here as well both comments.
```suggestion
                    background: showOnlyFollowingPosts ? "#FF6B08" : "",
```